snapcraft/parts/plugins/kernel.py

```diff
@@ -0,0 +1,1750 @@
+        # determine type of initrd
+        snapd_snap_file_name = _SNAPD_SNAP_FILE.format(
+            snap_name=_SNAPD_SNAP_NAME,
+            architecture=self._part_info.project_info.target_arch,
+        )
+
+        self.snapd_snap = os.path.join("${CRAFT_PART_BUILD}", snapd_snap_file_name)
+
+    def _get_kernel_architecture(self) -> None:
+        if self._part_info.project_info.target_arch == "armhf":
+            self.kernel_arch = "arm"
+        elif self._part_info.project_info.target_arch == "arm64":
+            self.kernel_arch = "arm64"
+        elif self._part_info.project_info.target_arch == "riscv64":
+            self.kernel_arch = "riscv"
+        elif self._part_info.project_info.target_arch == "amd64":
+            self.kernel_arch = "x86"
+        else:
+            logger.error("Unknown kernel architecture!!!")
+
+    def _get_deb_architecture(self) -> None:
+        if self._part_info.project_info.target_arch == "armhf":
+            self.deb_arch = "armhf"
+        elif self._part_info.project_info.target_arch == "arm64":
+            self.deb_arch = "arm64"
+        elif self._part_info.project_info.target_arch == "riscv64":
+            self.deb_arch = "riscv64"
+        elif self._part_info.project_info.target_arch == "amd64":
+            self.deb_arch = "amd64"
+        else:
+            logger.error("Unknown deb architecture!!!")
+
+    def _check_cross_compilation(self) -> None:
+        if self._cross_building:
+            self.make_cmd.append(f"ARCH={self.kernel_arch}")
+            self.make_cmd.append("CROSS_COMPILE=${CRAFT_ARCH_TRIPLET}-")
+
+    def _set_kernel_targets(self) -> None:
+        if not self.options.kernel_image_target:
+            self.kernel_image_target = _default_kernel_image_target[self.deb_arch]
+        elif isinstance(self.options.kernel_image_target, str):
+            self.kernel_image_target = self.options.kernel_image_target
+        elif self.deb_arch in self.options.kernel_image_target:
+            self.kernel_image_target = self.options.kernel_image_target[self.deb_arch]
+
+        self.make_targets = [self.kernel_image_target, "modules"]
+        self.make_install_targets = [
+            "modules_install",
+            "INSTALL_MOD_STRIP=1",
+            "INSTALL_MOD_PATH=${CRAFT_PART_INSTALL}",
+        ]
+        if self.options.kernel_device_trees:
+            self.dtbs = [f"{i}.dtb" for i in self.options.kernel_device_trees]
+            if self.dtbs:
+                self.make_targets.extend(self.dtbs)
+        elif self.kernel_arch in ("arm", "arm64", "riscv64"):
+            self.make_targets.append("dtbs")
+            self.make_install_targets.extend(
+                ["dtbs_install", "INSTALL_DTBS_PATH=${CRAFT_PART_INSTALL}/dtbs"]
+            )
+        self.make_install_targets.extend(self._get_fw_install_targets())
+
+    def _set_llvm(self) -> None:
+        if self._llvm_version is not None:
+            self.make_cmd.append(f'LLVM="{self._llvm_version}"')
```

Comment:
For clarity, it could be better to avoid calling a method just to set an attribute based on another attribute and set them directly in the caller instead.